src/Utilities/hooks/useBulkSelectConfig.js

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable camelcase */
+import { useDispatch, useStore } from 'react-redux';
+import * as actions from '../../store/actions';
+import React, { useCallback, useRef } from 'react';
+import useFetchBatched from './useFetchBatched';
+import { flatten, map } from 'lodash';
+import { loadSystems } from '../sharedFunctions';
+import { inventoryCache } from '../../components/InventoryTable/InventoryTable';
+import { Spinner } from '@patternfly/react-core';
+
+export const useBulkSelectConfig = (selected, filters, total, rows, loaded) => {
+    const [isBulkLoading, setBulkLoading] = React.useState(false);
+    const { fetchBatched } = useFetchBatched();
+    const dispatch = useDispatch();
+    const store = useStore();
+    const cache = useRef(inventoryCache());
+    const onSelectRows = (id, isSelected) => dispatch(actions.selectEntity(id, isSelected));
+    const calculateSelected = () => selected ? selected.size : 0;
+    const getEntitiesWrapper = async (filters, { page, per_page }) => {
+        const { activeFilters } = store.getState().entities;
+        const cachedProps = cache.current?.getProps() || {};
+
+        const newParams = {
+            page,
+            per_page,
+            items: cachedProps.items,
+            sortBy: cachedProps.sortBy,
+            hideFilters: cachedProps.hideFilters,
+            filters: activeFilters,
+            hasItems: cachedProps.hasItems,
+            //RHIF-246: Compliance app depends on activeFiltersConfig to apply its filters.
+            activeFiltersConfig: cachedProps.activeFiltersConfig
+        };
+        let resp = loadSystems(
+            { ...newParams },
+            cachedProps.showTags,
+            cachedProps.getEntities
+        );
+
+        const data = await resp.payload;
+        return data;
+
+    };
+
+    const calculateChecked = (rows = [], selected) => (
+        rows.every(({ id }) => selected && selected.has(id))
+            ? rows.length > 0
+            : rows.some(({ id }) => selected && selected.has(id)) && null
+    );
+    const fetchAllSystemIds = useCallback((filters, total) => {
+        return fetchBatched(getEntitiesWrapper, total, filters);
+    }, []);
+
+    const selectAllIds = async (selected = true) => {
+        setBulkLoading(true);
+        const data = await fetchAllSystemIds(filters, total);
+        const results = flatten(map(data, 'results'));
+        dispatch(actions.selectEntity(results, selected));
+        setBulkLoading(false);
+    };
+
+    return (
+        {
+            id: 'bulk-select-systems',
+            items: [{
+                title: 'Select none (0)',
```

Comment:
can we disable this item when there are no items selected? you can inspire by https://github.com/RedHatInsights/insights-inventory-frontend/blob/master/src/components/GroupSystems/GroupSystems.js#L20